assets/optin.go

```diff
@@ -0,0 +1,65 @@
+package assets
+
+import (
+	"fmt"
+
+	"github.com/nyaruka/gocommon/uuids"
+)
+
+// OptInUUID is the UUID of an opt in
+type OptInUUID uuids.UUID
+
+// OptIn are channel specific opt-ins
+//
+//	{
+//	  "uuid": "8925c76f-926b-4a63-a6eb-ab69e7a6b79b",
+//	  "name": "Joke Of The Day",
+//	  "channel": {
+//	     "uuid": "204e5af9-42c3-4d46-8aab-ce204dff25b4",
+//	     "name": "Facebook"
```

Comment:
making optins channel specific.. because we're going to associate auth keys with optins and they are obviously channel specific